lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -297,29 +298,56 @@ def grouper(iterable, chunk_size=self.USER_BATCH_SIZE, fillvalue=None):
         users = users.select_related('profile__allow_certificate')
         return grouper(users)
 
-    def _user_grade_results(self, course_grade, context):
+    def _user_grades(self, course_grade, context):
         """
         Returns a list of grade results for the given course_grade corresponding
         to the headers for this report.
         """
         grade_results = []
         for assignment_type, assignment_info in context.graded_assignments.iteritems():
-            for subsection_location in assignment_info['subsection_headers']:
-                try:
-                    subsection_grade = course_grade.subsection_grade(subsection_location)
-                except KeyError:
-                    grade_result = u'Not Available'
-                else:
-                    if subsection_grade.attempted_graded:
-                        grade_result = subsection_grade.graded_total.earned / subsection_grade.graded_total.possible
-                    else:
-                        grade_result = u'Not Attempted'
-                grade_results.append([grade_result])
-            if assignment_info['separate_subsection_avg_headers']:
-                assignment_average = course_grade.assignment_average(assignment_type)
+
+            subsection_grades, subsection_grades_results = self._user_subsection_grades(
+                course_grade,
+                assignment_info['subsection_headers'],
+            )
+            grade_results.extend(subsection_grades_results)
```

Comment:
This doesn't change to `extend` doesn't change the format of the end report at all?

Reply:
Good question.
The [code below](https://github.com/edx/edx-platform/pull/16170/files#diff-7683b9bc4a5398602e597d135a5af15bR334) in `user_subsection_grades` still does an append for the individual grade result cell-value.  The `extend` here then merges the list generated by `subsection_grades_results` with the `grade_results` list in this method.

Eventually, all these values get flattened with the `_flatten` call at the end of this method.  So I don't know why we append individual lists - just then to flatten them at the end.  But I'll leave that exploration for another PR.